MarsRoverBestGroup3.0/MarsRoverBestGroup3.0/APICall.cs

```diff
@@ -11,28 +11,36 @@ namespace MarsRoverBestGroup3._0
 {
     public class APICall
     {
-        public static IRestResponse GenericCall(string rest_request)
+        public static APOD AstronomyPhotoOfTheDay() 
         {
             var apiKey = ConfigurationManager.AppSettings["APIKey"];
             var client = new RestClient("https://api.nasa.gov");
-            var request = new RestRequest($"{rest_request}api_key={apiKey}", DataFormat.Json);
+            var request = new RestRequest($"/planetary/apod?api_key={apiKey}", DataFormat.Json);
             var response = client.Get(request);
             if (response.StatusCode != System.Net.HttpStatusCode.OK)
             {
                 throw new Exception("API call invalid");
             }
-            return response;
-        }
-        public static APOD AstronomyPhotoOfTheDay() 
-        {
-            var response = GenericCall("/planetary/apod?");
             var apod = JsonConvert.DeserializeObject<APOD>(response.Content);
             return apod;
         }
-        public static List<Photo> GetMarsRoverPhotosByDate(DateTime earth_date)
+        public static List<Photo> GetMarsRoverPhotosByDateAndRover(DateTime earth_date, string rover_name = "curiosity", string camera_name = null)
         {
             var earth_date_to_string = earth_date.ToString("yyyy-MM-dd");
-            var response = GenericCall($"/mars-photos/api/v1/rovers/curiosity/photos?earth_date={earth_date_to_string}&");
+            var api_key = ConfigurationManager.AppSettings["APIKey"];
+            var client = new RestClient("https://api.nasa.gov");
+            var request = new RestRequest($"/mars-photos/api/v1/rovers/{rover_name}/photos", DataFormat.Json);
+            request.AddParameter("earth_date", earth_date_to_string);
+            if (camera_name is not null)
+            {
+                request.AddParameter("camera", camera_name);
+            }
+            request.AddParameter("api_key", api_key);
+            var response = client.Get(request);
+            if (response.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("API call invalid");
+            }
             var mars_rover_pictures = JsonConvert.DeserializeObject<MarsRoverPhotosResponse>(response.Content);
```

Comment:
you should be able to get RestSharp to do this for you.

```csharp
var response = client.Get<MarsRoverPhotosResponse>(request);
return response.Data.Photos;
```

Reply:
Managed to get this working with the GetMarsRoverPhotosByDateAndRover method, although no luck with the AstronomyPhotoOfTheDay method